src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -128,8 +125,18 @@ export class KernelLauncher implements IKernelLauncher {
         @inject(IFileSystem) private file: IFileSystem
     ) {}
 
-    public async launch(interpreterUri: InterpreterUri, kernelName?: string): Promise<IKernelProcess> {
-        const kernelSpec = await this.kernelFinder.findKernelSpec(interpreterUri, kernelName);
+    public async launch(
+        interpreterUri: InterpreterUri,
+        kernelName?: string | IJupyterKernelSpec
+    ): Promise<IKernelProcess> {
+        let kernelSpec: IJupyterKernelSpec;
+        if (typeof kernelName === 'object') {
+            // IJupyterKernelSpec
+            kernelSpec = kernelName;
+        } else {
+            // string or undefined
+            kernelSpec = await this.kernelFinder.findKernelSpec(interpreterUri, kernelName);
+        }
```

Comment:
```suggestion
        if (typeof kernelName === 'string') {
            // string or undefined
            kernelSpec = await this.kernelFinder.findKernelSpec(interpreterUri, kernelName);
        } else {
            // IJupyterKernelSpec
            kernelSpec = kernelName;
        }
```

The reason is simple, `typeof null == 'object'`.
If we end up passing `null` for the value `kernelName`, then it will go into the wrong path.
Its a js thing, checking for object needs to be preceeded with checking for truthy...

Reply:
Nice! Didn't know that. Added the change also with a !kernelName on the first if so that if no kernel spec is defined it also goes down the findKernelSpec path.